### PR TITLE
Mouton styling

### PR DESCRIPTION
**Pull request by AlexMouton**

### CODE REVIEW

**Comment by MonoidMusician** [comment]
My other (kind of far off) thought with the plusles was to turn them into basically a small `<hr>` with a vertical stroke in the middle, but I have no idea how easy that would be to accomplish 😅 (beyond my CSS skills for sure).

**Comment by AlexMouton** [comment]
Thats a good step, I don't know that its even necessary though, ultimately.

File: docs/index.html [hunk 1]
```diff
@@ -39,62 +47,77 @@
...
+      }
+
+      input.constructor {
+        font-weight: 800;
+        margin-right: 15px;
       }
+
+      .type {
+        position: relative;
+      }
```
Comment (MonoidMusician) [inline docs/index.html]: Sorry I got uncreative and left this class (and `.constructor`) on both the `<div>` and the `<input>`, which one do you want this rule to apply to? ^^

**Comment by AlexMouton** [comment]
That should be on the div, to allow delete to be absolutely positioned

File: docs/index.html [hunk 2]
```diff
@@ -72,6 +72,7 @@
         overflow: auto;
         -webkit-overflow-scrolling: touch;
         white-space: nowrap;
+        display: inline-block;j
```
Comment (MonoidMusician) [inline docs/index.html]: you probably need this (only) on the child divs though?

**Comment by AlexMouton** [comment]
ah yeah, I guess I havnt seen how there are two levels of divs getting
'topClass' or is that itemClass for type items?
Probably at that threshold where I start doing negative good for tonight.

**Comment by MonoidMusician** [comment]
Looking great! Can we (and by we, I mean you – I have tried and failed! to) get `div.constructor.dragged, div.type.dragged` to obscure the element it is dragged over? I think it requires a background color and a higher z-index, but things didn't go well with my experiments with that ... in particular the remove buttons seem to hover a lot, and I ended up with little patches of background color too.

Also not convinced I like the phantom input, I'll add a single button back in at the end

**Comment by MonoidMusician** [comment]
Oh wow I can edit your comment.

Anyways, yeah, the component itself has a `div` (which should get `topClass`), and then it renders each child it its own div (which will be either a button for adding or an actual item ...), probably good to have it's own class.

In particular the parent div needs `overflow: auto`, and horizontal children will need to be `inline-block`.

**Comment by MonoidMusician** [comment]
Okay, I'm going to merge what we have now and move it over to `datagen`! In the future I should move the demos to the `test` folder ...

**Pull request merged**